### ISSUE TITLE
How to handle HTTP/2 GOAWAY ?

### ISSUE THREAD

**Issue by Raphael0010** [sindresorhus/got#2156]
#### What would you like to discuss?

Hi, I'm looking a way to handle this : "New streams cannot be created after receiving a GOAWAY".
I have read the RFC7540 about the GOAWAY frame, but I'm questioning if they exist a way to handle it with got ?

According to the RFC : `
To deal with this
   case, the GOAWAY contains the stream identifier of the last peer-
   initiated stream that was or might be processed on the sending
   endpoint in this connection.
`. 

So I should re-open the connection with this last stream identifier, but how I can get it with got ?
#### Checklist

- [X] I have read the documentation.

**Comment by szmarczak** [comment]
> I should re-open the connection with this last stream identifier

You don't. You create a new stream. IOW you just call `got` as usual.

**Comment by prabhatmishra-metro** [comment]
Hi @szmarczak @Raphael0010,
I get the same error, can you please tell me how to deal with it? 

Do I need to simply retry the request and everything would just work fine?

**Comment by prabhatmishra33** [comment]
Hi @szmarczak  @Raphael0010 ,
I get the same error, can you please tell me how to deal with it?

Do I need to simply retry the request and everything would just work fine?

**Comment by yovanoc** [comment]
We don't know. I have the same problem too but @szmarczak close this issue abruptly. I think too is not normal that simple requests fails like that

**Comment by mkaufmaner** [comment]
@Raphael0010 @prabhatmishra33 @yovanoc @szmarczak 

I ended up running into this error as mentioned in the https://github.com/szmarczak/http2-wrapper/issues/89 bug. I ended up forking the `http2-wrapper` library, change the following files https://github.com/szmarczak/http2-wrapper/pull/90/files , and pass the new agent in the got `request` option.

I was able to validate my fix through e2e testing but I still need to write an integration test for it.
